src/Android/Accessibility/AccessibilityHelpers.cs

```diff
@@ -330,15 +330,64 @@ public static void FillEditText(AccessibilityNodeInfo editTextNode, string value
                 if (KnownUsernameFields.ContainsKey(uriKey))
                 {
                     var usernameField = KnownUsernameFields[uriKey];
-                    if (uriLocalPath.EndsWith(usernameField.UriPathEnd))
+                    (string UriPathWanted, string UsernameViewId)[] accessOptions = usernameField.AccessOptions;
+
+                    for (int i = 0; i < accessOptions.Length; i++)
                     {
-                        foreach (var editText in allEditTexts)
+                        string curUriPathWanted = accessOptions[i].UriPathWanted;
+                        string curUsernameViewId = accessOptions[i].UsernameViewId;
+                        bool uriLocalPathMatches = false;
+
+                        // Case-sensitive comparison
+                        if (curUriPathWanted.StartsWith("startswith:", StringComparison.Ordinal))
+                        {
+                            curUriPathWanted = curUriPathWanted.Substring(11);
+                            uriLocalPathMatches = uriLocalPath.StartsWith(curUriPathWanted, StringComparison.Ordinal);
+                        }
+                        else if (curUriPathWanted.StartsWith("contains:", StringComparison.Ordinal))
+                        {
+                            curUriPathWanted = curUriPathWanted.Substring(9);
+                            uriLocalPathMatches = uriLocalPath.Contains(curUriPathWanted, StringComparison.Ordinal);
```

Comment:
Why use `.Contains()` here but below use `.IndexOf() >= 0` for the same check (given explicit passing of `StringComparison` mode to each method?

Reply:
Fixed. Thanks. :thumbsup: